cmake/kokkos_test_cxx_std.cmake

```diff
@@ -35,9 +35,18 @@ FUNCTION(kokkos_set_cxx_standard_feature standard)
     GLOBAL_SET(KOKKOS_CXX_STANDARD_FEATURE "")
   ENDIF()
 
-  IF(NOT ${FEATURE_NAME} IN_LIST CMAKE_CXX_COMPILE_FEATURES)
-    MESSAGE(FATAL_ERROR "Compiler ${KOKKOS_CXX_COMPILER_ID} should support ${FEATURE_NAME}, but CMake reports feature not supported")
-  ENDIF()
+    IF(NOT ${FEATURE_NAME} IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+          IF (KOKKOS_CXX_COMPILER_ID STREQUAL "NVIDIA")
+                  MESSAGE(STATUS "nvcc_wrapper does not support TARGET_COMPILE_FEATURES")
+                        GLOBAL_SET(KOKKOS_CXX_STANDARD_FEATURE "")
```

Comment:
I don't understand how this branch got brought back in. nvcc_wrapper should work now. Were there issues happening that required this change?

Reply:
`kokkos:incremental-testing` branch should not be touching that file at all.  We will need to do some history cleaning...

Reply:
A diff between https://github.com/kokkos/kokkos/pull/2633 and `develop` shows that the current changes are pretty self-contained and reasonably small. I guess we just rewrite the history for `incremental-testing` completely before merging into `develop`. At least, rebasing is a nightmare.

Reply:
I revert all those changes in #2633